android/java/org/chromium/chrome/browser/settings/developer/BraveQAPreferences.java

```diff
@@ -186,10 +187,9 @@ public void onStop() {
     public boolean onPreferenceChange(Preference preference, Object newValue) {
         if (PREF_USE_REWARDS_STAGING_SERVER.equals(preference.getKey())) {
             BraveRewardsNativeWorker.getInstance().ResetTheWholeState();
-            BravePrefServiceBridge.getInstance().setUseRewardsStagingServer((boolean) newValue);
+            mUseRewardsStagingServer = (boolean) newValue;
             mMaximizeAdsNumber.setEnabled((boolean) newValue);
             enableMaximumAdsNumber(((boolean) newValue) && mMaximizeAdsNumber.isChecked());
-            BraveRelaunchUtils.askForRelaunch(getActivity());
```

Comment:
Does it work without relaunch now?

Reply:
No, it's just that the `OnResetTheWholeState` is already asking for relaunch so it's not necessary to do it here too.